### DISCUSSION TITLE
Multiple search

### DISCUSSION THREAD

**Discussion by vasnt** [khoj-ai/khoj#121] [Q&A] [answered]
Hi, 

Is it possible to extend this search tool so that 
1) It can search multiple words (from a given set) in parallel i.e. "elon", "musk", "elon musk" or "elonmusk" (kind of binary OR / AND /NOT logic) 
2) stores results in a different  file with backlinks to original text from wherever it finds information.

**Comment by debanjum** [comment, marked as answer]
Hi @vasnt, thanks for opening a discussion topic for this idea!
What is the use-case you are imagining? Is this for use with Emacs, Obsidian or as an independent tool/script?

> i.e. "elon", "musk", "elon musk" or "elonmusk" (kind of binary OR / AND /NOT logic)

In the above example are you expecting the results to contain exact words matches?
If so, khoj supports [specifying specific words to include/exclude](https://github.com/debanjum/khoj#query-filters) but it only supports the `and` operations currently. For  e.g `+"elon" -"twitter"` will return all entries contain the word `elon` AND not containing the word `twitter`

The primary use-case for khoj is searching with natural language queries. For e.g `Elon musk's vision for Tesla". There are better tools like [org-search-view](https://orgmode.org/worg/org-tutorials/advanced-searching.html#keyword-searches) and [org-ql](https://github.com/alphapapa/org-ql#comparison-with-org-agenda-searches) in Emacs for exact word searches.

> stores results in a different file with backlinks to original text from wherever it finds information.

The org-mode results returned by Khoj contain 3 different links to the original text in [their property drawer](https://orgmode.org/manual/Property-Syntax.html), 1. line no, 2. heading, 3. org-id (when available). These can be used to jump to the original entries/headings[^1]
- If you're trying to do this manually from Emacs, you can just save the khoj results buffer as a new file. The results already link to their original entry
- If you're trying to do this programmatically, Khoj exposes an API[^2] from which you can fetch results and store in a different file. E.g `curl 'localhost:8000/api/search?q="What%20is%20the%20meaning%20of%20life"&t=org&n=2' | jq '.[].entry'`

[^1]:  By hitting the `o 2` [speed key](https://orgmode.org/manual/Speed-Keys.html) or `C-x C-o 2` or `M-x org-open-at-point RET 2`
[^2]: See the Khoj [FastAPI](https://fastapi.tiangolo.com/) [Swagger Docs](http://localhost:8000/docs) or [ReDocs](http://localhost:8000/redocs)
